lib/chef/knife/s3_source.rb

```diff
@@ -20,11 +20,11 @@ def bucket_obj
       end
 
       def bucket
-        URI(@url).host
+        URI(@url).path.split("/")[1]
```

Comment:
The valid s3 URL format in this case would be `s3://chef/encrypted_data_bag_secret`.